tests/50-sd-verify.js

```diff
@@ -23,147 +22,158 @@ const {match} = endpoints.filterByTag({
+          missingElements: []
+        }
+      }
+    };
+    before(async function() {
+      const {subjectNestedObjects, subjectHasArrays} = credentials.verify;
+      // create initial signed VCs
+      testVectors.signed = await issueTestData({
+        credential: subjectNestedObjects.document,
+        suite,
+        keyTypes,
+        mandatoryPointers: subjectNestedObjects.mandatoryPointers
+      });
+      const signedVc = testVectors.signed.get(keyTypes[0]);
+      // use initial VCs for a basic selective disclosure test
+      testVectors.disclosed.base = await deriveTestData({
+        selectivePointers: ['/credentialSubject/id'],
+        verifiableCredential: signedVc,
+        keyTypes,
+        suite
+      });
+      // create initial nestedDisclosedCredential from signedVc
+      testVectors.disclosed.nested = await deriveTestData({
+        selectivePointers: subjectNestedObjects.selectivePointers.slice(1, 3),
+        verifiableCredential: signedVc,
+        keyTypes,
+        suite
+      });
+      // copy the first vc
+      const noIdVc = klona(subjectNestedObjects.document);
+      // delete the id
+      delete noIdVc.id;
+      // start second round test data creation w/ dlCredentialNoIds
+      const noIdsVcs = await issueTestData({
+        credential: noIdVc,
+        keyTypes,
+        suite: 'ecdsa-sd-2023',
+        mandatoryPointers: subjectNestedObjects.mandatoryPointers
+      });
+      const signedDlCredentialNoIds = noIdsVcs.get(keyTypes[0]);
+      testVectors.disclosed.noIds = await deriveTestData({
+        selectivePointers: subjectNestedObjects.selectivePointers.slice(1, 3),
+        verifiableCredential: signedDlCredentialNoIds,
+        keyTypes,
+        suite
+      });
+      const credentialHasArrays = klona(subjectHasArrays);
+      // start third round test data creation w/
+      // AchievementCredential
+      const achievementCredentials = await issueTestData({
+        credential: credentialHasArrays.document,
+        mandatoryPointers: credentialHasArrays.mandatoryPointers,
+        keyTypes,
+        suite
+      });
+      const signedAchievementCredential = achievementCredentials.get(
+        keyTypes[0]);
+      // select full arrays
+      testVectors.disclosed.array.full = await deriveTestData({
+        selectivePointers:
+          [...credentialHasArrays.selectivePointers],
+        verifiableCredential: signedAchievementCredential,
+        suite,
+        keyTypes
+      });
+      // select less than full subarrays
+      const lessThanFullPointers = credentialHasArrays.
+        selectivePointers.slice(2, -4);
+      testVectors.disclosed.array.lessThanFull = await deriveTestData({
+        selectivePointers: lessThanFullPointers,
+        verifiableCredential: signedAchievementCredential,
+        suite,
+        keyTypes
+      });
+      // select w/o first 7 array element
+      const removeFirst7Pointers = credentialHasArrays.
+        selectivePointers.slice(7);
```

Comment:
What is the significance of `7`?

Reply:
I asked the same thing before...I don't think it's significant...but since it keeps coming up, we should get it explained in a comment at least.

Reply:
these are ports of original test data so the idea was to just port them in a programmatic way that cut down on the number of test vectors needed to create the test data. Originally this test vector and the one below it were identical for reasons not provided by the original dev. I think what we really want is for @Wind4Greg to provide some guidance on what the sd verifier test inputs should look like and then let him provide them as he is more of an expert in the sd field and than any of us are. @Wind4Greg do you think you could revise this test data in your fork at a later point?

Reply:
Hi all and @aljones15 in particular. I'll be updating the spec test vectors with something more relevant that my windsurfing examples. However, the longer examples that you came up with (whether they made application sense or not) helped me find a particularly subtle error in my SD primitive code so I think we'll want something based on what's in the spec (to be updated) and something longer that pushes things a bit more like what you came up with. Maybe something that looks like an education transcript without all the details of a CLR (comprehensive learner record).